src/bootstrap/bootstrap.py

```diff
@@ -1057,7 +1076,25 @@ def bootstrap(args):
     # Fetch/build the bootstrap
     build.download_toolchain()
     sys.stdout.flush()
-    build.build_bootstrap(args.color, args.warnings, verbose_count)
+
+
+    try:
+        if not did_path_changed("src/bootstrap"):
```

Comment:
this path will not work when `x` called from outside of the rust tree.

Reply:
What could be a better way to check path changes I think there was some work done by you in a PR about bootstrap-shim binary but I could not find the PR?

Reply:
> What could be a better way to check path changes

Checking with git is completely fine, but `src/bootstrap` is a relative path and will not work when there is no `src/bootstrap` in the current working directory. There are many people(including me) using `x` from outside of the rust tree.

> I think there was some work done by you in a PR about bootstrap-shim binary but I could not find the PR?

Here https://github.com/rust-lang/rust/pull/107812/files#diff-1e2a2bd13194bee1cb25532bfe2cd41f9d7bf25ca62f42f32ea302d97de71356R55-R79

Reply:
> Here [#107812 (files)](https://github.com/rust-lang/rust/pull/107812/files#diff-1e2a2bd13194bee1cb25532bfe2cd41f9d7bf25ca62f42f32ea302d97de71356R55-R79)

where is config.last_modified_commit defined? The config.rs file doesn't seem to contain the function. I also did a code search and still couldn't find the function.

Reply:
> where is config.last_modified_commit defined? The config.rs file doesn't seem to contain the function. I also did a code search and still couldn't find the function.

https://github.com/ozkanonur/rust/blob/8ad89068e2558940b4184d46763ff9c6b325883b/src/bootstrap/min_config.rs#L178-L233

Reply:
I changed the logic to first get the toplevel and also started passing cwd to the Popen.